packages/nx/src/hasher/task-hasher.ts

```diff
@@ -429,22 +445,30 @@ class TaskHasherImpl {
         partialHashes.push(node.data.version);
       }
       // we want to calculate the hash of the entire dependency tree
-      if (this.projectGraph.dependencies[projectName]) {
-        this.projectGraph.dependencies[projectName].forEach((d) => {
-          if (!visited.has(d.target)) {
-            partialHashes.push(this.hashExternalDependency(d.target, visited));
-          }
-        });
-      }
+      this.projectGraph.dependencies[targetProjectName]?.forEach((d) => {
```

Comment:
Please keep the original format as optional chaining adds additional unnecessary checks when compiled to javascript. Also it's a bit more readable in my opinion.